packages/swapper/src/swappers/RelaySwapper/utils/relayTokenToAssetId.ts

```diff
@@ -0,0 +1,90 @@
+import type { AssetId, ChainReference } from '@shapeshiftoss/caip'
+import {
+  ASSET_NAMESPACE,
+  ASSET_REFERENCE,
+  CHAIN_NAMESPACE,
+  CHAIN_REFERENCE,
+  toAssetId,
+  toChainId,
+} from '@shapeshiftoss/caip'
+
+import { DEFAULT_RELAY_EVM_TOKEN_ADDRESS } from '../constant'
+import type { RelayToken } from './types'
+
+export const relayTokenToAssetId = (relayToken: RelayToken): AssetId => {
+  const chainReference = relayToken.chainId.toString() as ChainReference
+  const chainId = toChainId({
+    chainNamespace: CHAIN_NAMESPACE.Evm,
+    chainReference,
+  })
+
+  // @TODO: Handle the same for Solana and BTC
+  const isDefaultAddress = relayToken.address === DEFAULT_RELAY_EVM_TOKEN_ADDRESS
+
+  const { assetReference, assetNamespace } = (() => {
+    // @TODO: Handle the same for Solana and BTC
+    if (!isDefaultAddress)
+      return {
+        assetReference: relayToken.address,
+        assetNamespace:
+          chainReference === CHAIN_REFERENCE.BnbSmartChainMainnet
+            ? ASSET_NAMESPACE.bep20
+            : ASSET_NAMESPACE.erc20,
+      }
+    switch (chainReference) {
+      case CHAIN_REFERENCE.EthereumMainnet:
+        return {
+          assetReference: ASSET_REFERENCE.Ethereum,
+          assetNamespace: ASSET_NAMESPACE.slip44,
+        }
+      case CHAIN_REFERENCE.AvalancheCChain:
+        return {
+          assetReference: ASSET_REFERENCE.AvalancheC,
+          assetNamespace: ASSET_NAMESPACE.slip44,
+        }
+      case CHAIN_REFERENCE.OptimismMainnet:
+        return {
+          assetReference: ASSET_REFERENCE.Optimism,
+          assetNamespace: ASSET_NAMESPACE.slip44,
+        }
+      case CHAIN_REFERENCE.PolygonMainnet:
+        return {
+          assetReference: ASSET_REFERENCE.Polygon,
+          assetNamespace: ASSET_NAMESPACE.slip44,
+        }
+      case CHAIN_REFERENCE.GnosisMainnet:
+        return {
+          assetReference: ASSET_REFERENCE.Gnosis,
+          assetNamespace: ASSET_NAMESPACE.slip44,
+        }
+      case CHAIN_REFERENCE.ArbitrumMainnet:
+        return {
+          assetReference: ASSET_REFERENCE.Arbitrum,
+          assetNamespace: ASSET_NAMESPACE.slip44,
+        }
+      case CHAIN_REFERENCE.BaseMainnet:
+        return {
+          assetReference: ASSET_REFERENCE.Base,
+          assetNamespace: ASSET_NAMESPACE.slip44,
+        }
+      case CHAIN_REFERENCE.BitcoinMainnet:
+        return {
+          assetReference: ASSET_REFERENCE.Bitcoin,
+          assetNamespace: ASSET_NAMESPACE.slip44,
+        }
+      case CHAIN_REFERENCE.SolanaMainnet:
+        return {
+          assetReference: ASSET_REFERENCE.Solana,
+          assetNamespace: ASSET_NAMESPACE.splToken,
+        }
```

Comment:
q: what about Solana native asset (slip44)? looks like this should belong in the early return, and slip44 here instead

Reply:
Yeah I fixed it in the solana's PR but happy to change it here also in even thought it doesn't make sense as Solana isn't implemented for now 

Changed in [7e3a115](https://github.com/shapeshift/web/pull/9188/commits/7e3a115c7aeb37843ca13844d6f4f73501a40f03)